### DISCUSSION TITLE
Prevent overflowing of links within lists or tables

### DISCUSSION THREAD

**Discussion by miliog** [typst/typst#2365] [Q&A] [answered]
I need something like `xurl` for Typst. How can I achieve this? I often have links that are longer than usual, and I don't want to shorten them in any way.

Exampe in LaTeX:

<img width="1202" alt="image" src="https://github.com/typst/typst/assets/146289769/d0cd3552-1852-4228-b7c4-f2a839a14ac5">

Non-working example in Typst:

```typ
- https://google.com/dskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdas

#table(columns: (auto, 1fr),
[*No.*], [*URL*],
[1], [https://example.com/dskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdas],
[2], [https://anotherexample.com/dskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdasdskalfjksdflkjfdsalfdas]
)
```
![Untitled](https://github.com/typst/typst/assets/146289769/57c49a51-0886-460f-90b7-844d89302675)

**Comment by emilyyyylime** [comment, marked as answer]
You can insert Zero width spaces in the `link` content.
One approach is to just go ham and intersperse every link with them:
```
#let softlink(url) = link(url, url.split("").slice(1,-1).join(str(sym.zws)))
#show regex("\w+://[\w-]+\.\S+"): match => { softlink(match.text) }
```

**Reply by miliog** [reply]
Thank you! Works perfectly from what I can tell!
